Framework/Backend/websocket/server.js

```diff
@@ -109,18 +109,19 @@ class WebSocket {
    */
   onconnection(client, request) {
     const token = url.parse(request.url, true).query.token;
-    this.http.jwt.verify(token)
-      .then((decoded) => {
-        client.id = decoded.id;
-        client.send(JSON.stringify({command: 'authed', id: client.id}));
-        client.on('message', (message) => this.onmessage(message, client));
-        client.on('close', () => this.onclose(client));
-        client.on('pong', () => client.isAlive = true);
-        client.on('error', (err) => this.log.error(`Connection ${err.code}`));
-      }, (error) => {
-        this.log.debug(`${error.name} : ${error.message}`);
-        client.close(1008);
-      });
+    try {
+      const decoded = this.http.o2WebToken.verify(token);
```

Comment:
I think the try should wrap as less code as possible, so here only this line. Then in the catch you return after the close, and you continue manipulating client after

Reply:
Good suggestion, updated accordingly.